api/utils/keys/yubikey.go

```diff
@@ -84,25 +71,40 @@ func getOrGenerateYubiKeyPrivateKey(touchRequired bool) (*PrivateKey, error) {
 		return nil
 	}
 
-	// Check the client certificate in the slot.
-	switch cert, err := y.getCertificate(pivSlot); {
-	case err == nil && (len(cert.Subject.Organization) == 0 || cert.Subject.Organization[0] != certOrgName):
-		// Unknown cert found, prompt the user before we overwrite the slot.
-		if err := promptOverwriteSlot(nonTeleportCertificateMessage(pivSlot, cert)); err != nil {
+	// If a specific slot was specified, use that. Otherwise, check for a key in the
+	// default slot for the given policy and generate a new one if needed.
+	var pivSlot piv.Slot
+	if slot != "" {
+		pivSlot, err = parsePIVSlotString(slot)
```

Comment:
nit: If you go with a new type for the `string` instead of this free function you could create a 
```go
func (p PIVSlot) Parse() (piv.Slot, error)
```